src/mixins/highlightConfig.js

```diff
@@ -55,52 +58,45 @@ export default {
   methods: {
     setHighlight() {
       if (this.highlighted) {
-        this.shapeView.highlight(this.shapeBody, {highlighter: defaultHighlighter});
+        this.shapeView.highlight(this.shapeBody, { highlighter: defaultHighlighter });
         return;
       }
 
-      this.shapeView.unhighlight(this.shapeBody, {highlighter: defaultHighlighter});
+      this.shapeView.unhighlight(this.shapeBody, { highlighter: defaultHighlighter });
     },
     setErrorHighlight() {
       if (!this.shapeView) {
         return;
       }
-
-      if (this.hasError) {
+      if (this.hasError && this.autoValidate) {
         this.shapeView.highlight(null, { highlighter: errorHighlighter });
-      } else {
-        this.shapeView.unhighlight(null, { highlighter: errorHighlighter });
+        return;
       }
+      this.shapeView.unhighlight(null, { highlighter: errorHighlighter });
     },
-    configureCrown() {
-      this.paper.once('render:done', () => this.setHighlight());
-
-      const shapeView = this.shape.findView(this.paper);
+    setUpHighlightConfig() {
+      this.setHighlight();
+      this.setErrorHighlight();
 
       this.shape.on('change:size', () => {
         if (this.highlighted) {
           /* Ensure the highlight box expands to fit element */
-          shapeView.unhighlight(this.shapeBody, { highlighter: defaultHighlighter });
-          shapeView.highlight(this.shapeBody, { highlighter: defaultHighlighter });
+          this.shapeView.unhighlight(this.shapeBody, { highlighter: defaultHighlighter });
+          this.shapeView.highlight(this.shapeBody, { highlighter: defaultHighlighter });
         }
       });
     },
-    setUpCrownConfig() {
-      this.configureCrown();
-
-      this.setErrorHighlight();
-    },
   },
   mounted() {
     this.$nextTick(() => {
       /* Use nextTick to ensure this code runs after the component it is mixed into mounts.
        * This will ensure this.shape is defined. */
 
-      if (this.isRendering) {
-        this.paper.once('render:done', this.setUpCrownConfig);
-      } else {
-        this.setUpCrownConfig();
+      if (this.paperManager.hasScheduledUpdates()) {
+        this.paperManager.addOnceHandler('render:done', this.setUpHighlightConfig());
```

Comment:
The 2nd argument for `addOnceHandler` needs to be a function reference, you are currently calling the method there instead.
```suggestion
        this.paperManager.addOnceHandler('render:done', this.setUpHighlightConfig);
```